src/acrcssc/azext_acrcssc/helper/_deployment.py

```diff
@@ -0,0 +1,150 @@
+
+
+def validate_template(cmd_ctx, resource_group, deployment_name, template):
+    # Validation is automatically re-attempted in live runs, but not in test
+    # playback, causing them to fail. This explicitly re-attempts validation to
+    # ensure the tests pass
+    api_clients = cf_resources(cmd_ctx)
+    validation_res = None
+    deployment = Deployment(
+        properties=template,
+        # tags = { "test": CSSC_TAGS }, #we need to know if tagging
+        # is something that will help ust, tasks are proxy resources,
+        # so not sure how that would work
+    )
+
+    for validation_attempt in range(2):
+        try:
+            validation = (
+                api_clients.deployments.begin_validate(
+                    resource_group_name=resource_group,
+                    deployment_name=deployment_name,
+                    parameters=deployment
+                )
+            )
+            validation_res = LongRunningOperation(
+                cmd_ctx, "Validating ARM template..."
+            )(validation)
+            break
+        except Exception:  # pylint: disable=broad-except
+            if validation_attempt == 1:
+                raise
+
+    if not validation_res:
+        # Don't expect to hit this but it appeases mypy
+        raise RuntimeError(f"Validation of template {template} failed.")
+
+    logger.debug(f"Validation Result {validation_res}")
+    if validation_res.error:
+        # Validation failed so don't even try to deploy
+        logger.error(
+            (
+                f"Template for resource group {resource_group} has failed validation. The message"
+                " was: {validation_res.error.message}. See logs for additional details."
+            )
+        )
+        logger.debug(
+            (
+                f"Template for resource group {resource_group} failed validation."
+                " Full error details: {validation_res.error}"
+            )
+        )
+        raise RuntimeError("Azure template validation failed.")
+
+    # Validation succeeded so proceed with deployment
+    logger.debug("Successfully validated resources for {resource_group}")
+
+
+def deploy_template(cmd_ctx, resource_group, deployment_name, template):
+    api_client = cf_resources(cmd_ctx)
+
+    deployment = Deployment(
+        properties=template,
+        # tags = { "test": CSSC_TAGS },
+        # we need to know if tagging is something that will help us,
+        # tasks are proxy resources, so not sure how that would work
+    )
+
+    poller = api_client.deployments.begin_create_or_update(
+        resource_group_name=resource_group,
+        deployment_name=deployment_name,
+        parameters=deployment)
+    logger.debug(poller)
+
+    # Wait for the deployment to complete and get the outputs
+    deployment: DeploymentExtended = LongRunningOperation(
+        cmd_ctx
+    )(poller)
+    logger.debug("Finished deploying")
+
+    if deployment.properties is not None:
```

Comment:
Suggestion: simply the condition check by removing the if-else block and directly throw the error if deployment.properties is None.
e.g.
if deployment.properties is None:
    raise RuntimeError("The deployment has no properties.\nAborting")

depl_props = deployment.properties